### PR TITLE
[stdlib] StringProtocol no longer refines RangeReplaceableCollection

### PR DESCRIPTION
**Pull request by moiseev**
https://bugs.swift.org/browse/SR-5379

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by shahmishal** [comment]
@swift-ci please test macOS

File: stdlib/public/core/StringRangeReplaceableCollection.swift.gyb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 //
 //===----------------------------------------------------------------------===//
 
-extension String : StringProtocol {
+extension String : StringProtocol, RangeReplaceableCollection {
```
Comment (airspeedswift) [inline stdlib/public/core/StringRangeReplaceableCollection.swift.gyb]: Do we need the same on `Substring`?

Comment (moiseev) [inline stdlib/public/core/StringRangeReplaceableCollection.swift.gyb]: [No](https://github.com/apple/swift/blob/master/stdlib/public/core/Substring.swift.gyb#L393)

**Review by airspeedswift — APPROVED** [review]

**Review by dabrahams — APPROVED** [review]
LGTM

**Pull request merged**